packages/siwe/src/mapToSIWX.ts

```diff
@@ -19,37 +21,51 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
     }
   }
 
-  ChainController.subscribeKey('activeCaipNetwork', async activeCaipNetwork => {
-    if (!siwe.options.signOutOnNetworkChange) {
-      return
-    }
+  async function signOut() {
+    await siwe.methods.signOut()
+    siwe.methods.onSignOut?.()
+  }
 
-    const session = await getSession()
-    const isDiffernetNetwork =
-      session &&
-      session.chainId !== NetworkUtil.caipNetworkIdToNumber(activeCaipNetwork?.caipNetworkId)
+  subscriptions.forEach(unsubscribe => unsubscribe())
+  subscriptions.push(
+    ChainController.subscribeKey('activeCaipNetwork', async activeCaipNetwork => {
+      if (!siwe.options.signOutOnNetworkChange) {
+        return
+      }
 
-    if (isDiffernetNetwork) {
-      await siwe.methods.signOut()
-    }
-  })
+      const session = await getSession()
+      const isDifferentNetwork =
+        session &&
+        session.chainId !== NetworkUtil.caipNetworkIdToNumber(activeCaipNetwork?.caipNetworkId)
 
-  ChainController.subscribeKey('activeCaipAddress', async activeCaipAddress => {
-    if (!siwe.options.signOutOnAccountChange) {
-      return
-    }
+      if (isDifferentNetwork) {
+        await signOut()
+      }
+    }),
+    ChainController.subscribeKey('activeCaipAddress', async activeCaipAddress => {
+      if (siwe.options.signOutOnDisconnect && !activeCaipAddress) {
+        const session = await getSession()
+        if (session) {
+          await signOut()
+        }
 
-    const session = await getSession()
+        return
+      }
 
-    const compareSessionAddress = session?.address.toLowerCase()
-    const compareCaipAddress = CoreHelperUtil?.getPlainAddress(activeCaipAddress)?.toLowerCase()
+      if (siwe.options.signOutOnAccountChange) {
```

Comment:
What if the param isn't provided. Based on docs it should default to `true`.
https://docs.reown.com/appkit/next/core/siwe#signoutondisconnect-boolean---defaults-to-true

Reply:
shouldn't this be set during initialization as a default value?